cpp/src/gandiva/precompiled/string_ops.cc

```diff
@@ -410,4 +410,78 @@ const char* convert_fromUTF8_binary(int64 context, const char* bin_in, int32 len
   return ret;
 }
 
+FORCE_INLINE
+const char* replace_utf8_utf8_utf8(int64 context, const char* text, int32 text_len,
```

Comment:
can we break this into two functions - one that accepts a max length as input and this one that just wraps the other one by passing in the max length..

that way we can test the inner function for max length errors too..